docs/overview.md

```diff
@@ -0,0 +1,39 @@
+# Shardy
```

Comment:
`overview.md` and `architecture.md` has many overlaps. Is it expected? Could we merge them into one?

Reply:
Yes I think after the last commit they are redundant. I will consolidate the content into overview.md